src/provider/provider_level_zero.c

```diff
@@ -343,7 +343,7 @@ static umf_result_t ze_memory_provider_alloc(void *provider, size_t size,
     ze_memory_provider_t *ze_provider = (ze_memory_provider_t *)provider;
 
     ze_result_t ze_result = ZE_RESULT_SUCCESS;
-    switch (ze_provider->memory_type) {
+    switch ((umf_usm_memory_type_t)ze_provider->memory_type) {
```

Comment:
If these are different enums, we should not do any casting - implicit or explicit.
Instead, we should add something like “ze2umf_memory_type” similar to already existing “ze2umf_result”